lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -531,45 +561,52 @@ def main():
                     rule['from_port'] = None
                     rule['to_port'] = None
 
-                # Convert ip to list we can iterate over
-                if not isinstance(ip, list):
-                    ip = [ip]
-
-                # If rule already exists, don't later delete it
-                for thisip in ip:
-                    ruleId = make_rule_key('out', rule, group_id, thisip)
+                if group_id:
+                    ruleId = make_rule_key('out', rule, group.id, group_id)
                     if ruleId in groupRules:
                         del groupRules[ruleId]
-                    # Otherwise, add new rule
                     else:
-                        grantGroup = None
-                        if group_id:
-                            grantGroup = groups[group_id].id
-
                         if not module.check_mode:
-                            ec2.authorize_security_group_egress(
-                                group_id=group.id,
-                                ip_protocol=rule['proto'],
-                                from_port=rule['from_port'],
-                                to_port=rule['to_port'],
-                                src_group_id=grantGroup,
-                                cidr_ip=thisip)
-                        changed = True
+                            ipPermissions = createGrantGroupIpPermission(group_id, rule)
+                            if ipPermissions:
+                                ips=ipPermissions
+                                if vpc_id:
+                                    ips=map(lambda x: map(lambda y: y.update({'VpcId': vpc_id}), x.get('UserIdGroupPairs')), [ipPermissions])
```

Comment:
This code is pretty incomprehensible, I'd try and avoid map and lambda and use a list/dictionary comprenhension (or a for loop if it's still unreadable)